binary function as parameter

### The following program `sample.go` triggers an unexpected result

```go
package main

import (
    "fmt"
    "math"
)   
    
type obj struct {
    num float64
}       
    
type Func func(o *obj) (r *obj, err error)

func numFunc(fn func(f float64) float64) Func {
    return func(o *obj) (*obj, error) {
        return &obj{fn(o.num)}, nil
    } 
}
  
func main() {
    f := numFunc(math.Cos)
    r, err := f(&obj{})
    fmt.Println(r, err)
}
```


### Expected result

```console
$ go run ./sample.go
&{1} <nil>
```


### Got

```console
$ yaegi ./sample.go
/Users/marc/yt/aj0.go:21:15: panic
run: reflect.Set: value of type func(float64) float64 is not assignable to type *interp.node
goroutine 1 [running]:
runtime/debug.Stack()
	runtime/debug/stack.go:24 +0x88
github.com/traefik/yaegi/interp.(*Interpreter).Execute.func1(0x14000255ae0)
	github.com/traefik/yaegi/interp/program.go:131 +0x8c
panic({0x102ebed40, 0x14000112940})
	runtime/panic.go:1038 +0x21c
github.com/traefik/yaegi/interp.runCfg.func1(0x1400034a160, 0x140001dad80, 0x0)
	github.com/traefik/yaegi/interp/run.go:193 +0x164
panic({0x102ebed40, 0x14000112940})
	runtime/panic.go:1038 +0x21c
reflect.Value.assignTo({0x102ecb2c0, 0x102fd06c8, 0x13}, {0x102cdaad2, 0xb}, 0x102fa0fe0, 0x0)
	reflect/value.go:2797 +0x2cc
reflect.Value.Set({0x102fa0fe0, 0x14000114368, 0x196}, {0x102ecb2c0, 0x102fd06c8, 0x13})
	reflect/value.go:1905 +0xa4
github.com/traefik/yaegi/interp.call.func9(0x1400034a160)
	github.com/traefik/yaegi/interp/run.go:1411 +0xb24
github.com/traefik/yaegi/interp.runCfg(0x140001dad80, 0x1400034a160, 0x140004cdc20, 0x0)
	github.com/traefik/yaegi/interp/run.go:201 +0x268
github.com/traefik/yaegi/interp.(*Interpreter).run(0x14000348000, 0x140004cdc20, 0x1400034a000)
	github.com/traefik/yaegi/interp/run.go:120 +0x3d4
github.com/traefik/yaegi/interp.(*Interpreter).Execute(0x14000348000, 0x14000400de0)
	github.com/traefik/yaegi/interp/program.go:157 +0x1e0
github.com/traefik/yaegi/interp.(*Interpreter).eval(0x14000348000, {0x140002c6dc0, 0x15c}, {0x16d9bb790, 0x15}, 0x0)
	github.com/traefik/yaegi/interp/interp.go:615 +0x98
github.com/traefik/yaegi/interp.(*Interpreter).EvalPath(0x14000348000, {0x16d9bb790, 0x15})
	github.com/traefik/yaegi/interp/interp.go:508 +0xe8
main.runFile(0x14000348000, {0x16d9bb790, 0x15}, 0x0)
	github.com/traefik/yaegi/cmd/yaegi/run.go:153 +0xb0
main.run({0x140001c0010, 0x1, 0x1})
	github.com/traefik/yaegi/cmd/yaegi/run.go:116 +0xe14
main.main()
	github.com/traefik/yaegi/cmd/yaegi/yaegi.go:144 +0x384
```


### Yaegi Version

0.11.2

### Additional Notes

_No response_

--- a/_test/issue-1361.go
+++ b/_test/issue-1361.go
@@ -9,16 +9,16 @@ type obj struct {
 	num float64
 }
 
-type Func func(o *obj) (r *obj, err error)
+type Fun func(o *obj) (r *obj, err error)
 
-func numFunc(fn func(f float64) float64) Func {
+func numFun(fn func(f float64) float64) Fun {
 	return func(o *obj) (*obj, error) {
 		return &obj{fn(o.num)}, nil
 	}
 }
 
 func main() {
-	f := numFunc(math.Cos)
+	f := numFun(math.Cos)
 	r, err := f(&obj{})
 	fmt.Println(r, err)
 }

--- a/_test/issue-1361.go
+++ b/_test/issue-1361.go
@@ -1,0 +1,27 @@
+package main
+
+import (
+	"fmt"
+	"math"
+)
+
+type obj struct {
+	num float64
+}
+
+type Func func(o *obj) (r *obj, err error)
+
+func numFunc(fn func(f float64) float64) Func {
+	return func(o *obj) (*obj, error) {
+		return &obj{fn(o.num)}, nil
+	}
+}
+
+func main() {
+	f := numFunc(math.Cos)
+	r, err := f(&obj{})
+	fmt.Println(r, err)
+}
+
+// Output:
+// &{1} <nil>

--- a/interp/run.go
+++ b/interp/run.go
@@ -1235,6 +1235,8 @@ func call(n *node) {
 				values = append(values, genValueInterface(c))
 			case isInterfaceBin(arg):
 				values = append(values, genInterfaceWrapper(c, arg.rtype))
+			case isFuncSrc(arg):
+				values = append(values, genValueNode(c))
 			default:
 				values = append(values, genValue(c))
 			}

--- a/interp/value.go
+++ b/interp/value.go
@@ -421,7 +421,11 @@ func genValueNode(n *node) func(*frame) reflect.Value {
 	value := genValue(n)
 
 	return func(f *frame) reflect.Value {
-		return reflect.ValueOf(&node{rval: value(f)})
+		v := value(f)
+		if _, ok := v.Interface().(*node); ok {
+			return v
+		}
+		return reflect.ValueOf(&node{rval: v})
 	}
 }
 

